File: 520-detect-capital/detect-capital.cs
public class Solution {
    public bool DetectCapitalUse(string word) {
        if((word.All(char.IsUpper)) || (word.All(char.IsLower)))
        {
            return true;
        }
        if(char.IsUpper(word[0]))
        {
            if(word.Substring(1).All(char.IsUpper)){
                return true;
            }
            else if(word.Substring(1).All(char.IsLower)){
                return true;
            }
            else{
                return false;
            }
        }
        if(char.IsLower(word[0]))
        {
            if(word.Substring(0).All(char.IsLower))
            {
                return true;
            }
            else{
                return false;
            }
        }
        return false;
    }
}

File: 520-detect-capital/detect-capital.py
class Solution(object):
    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """
        if word.isupper() or word.islower():
            return True
        if word[0].isupper():
            if word[1:].isupper():
                return True
            elif word[1:].islower():
                return True
            else:
                return False
        if word[0].islower():
            if word[1:].islower():
                return True
            else:
                return False